File: moderate/reverse_and_add.py
#!/usr/bin/env python2
# CodeEval - Reverse and Add
# https://www.codeeval.com/open_challenges/45/

from sys import argv

def reverse_add(interger_str):
    reverse_str = interger_str[::-1]
    sum_ = int(interger_str) + int(reverse_str)
    return str(sum_)


def is_palindrome(interger_str):
    if interger_str == interger_str[::-1]:
        return True
    else:
        return False


def main():
    f = open(argv[1], "rt")
    for line in f:
        interger_str = line.strip()
        iterations = 0
        while True:
            interger_str =  reverse_add(interger_str)
            iterations += 1
            if is_palindrome(interger_str):
                print("%d %s" % (iterations, interger_str))
                break

    f.close()


if __name__ == "__main__":
    main()

File: moderate/reverse_and_add.cs
/* CodeEval - Reverse and Add
 * https://www.codeeval.com/open_challenges/45/
 */

using System.IO;
using System.Collections.Generic;

class Program
{
    static string ReverseString(string someString)
    {
        char[] reversedCharArr = new char[someString.Length];
        int j = 0;
        for (int i = someString.Length-1; i >= 0; i--)
        {
            reversedCharArr[j] = someString[i];
            j++;
        }
        string reversedString = string.Join("" , reversedCharArr);
        return reversedString;
    }

    static string ReverseAdd(string intString)
    {
        string reversedString = ReverseString(intString);
        uint sum = uint.Parse(intString) + uint.Parse(reversedString);
        return sum.ToString();
    } 

    static bool IsPalindrome(string intString)
    {
        if (intString == ReverseString(intString))
            return true;
        else
            return false;
    }

    static void Main(string[] args)
    {
        using (StreamReader reader = File.OpenText(args[0]))
        while (!reader.EndOfStream)
        {
            string line = reader.ReadLine();
            if (null == line)
                continue;
            string intString = line.Trim();
            int iterations = 0;
            while (true)
            {
                intString = ReverseAdd(intString);
                //System.Console.WriteLine(intString);
                iterations++;
                if (IsPalindrome(intString))
                {
                    System.Console.WriteLine(iterations + " " + intString);
                    break;
                }
            }
        }
    }
}
